src/gridcoin/staking/kernel.cpp

```diff
@@ -499,13 +499,36 @@ uint256 GRC::CalculateStakeHashV8(
     return ss.GetHash();
 }
 
+uint256 GRC::CalculateStakeHashV8(
+    unsigned int nBlockTime,
+    const CTransaction& CoinTx,
+    unsigned CoinTxN,
+    unsigned nTimeTx,
+    uint64_t StakeModifier)
+{
+    CHashWriter ss(SER_GETHASH, 0);
+
+    ss << StakeModifier;
+    ss << (uint32_t) nBlockTime;
```

Comment:
Is it okay for this to be not masked?

Reply:
Yes. That is not the same as the transaction time, nTimeTx, which is masked.

Reply:
These optimizations on the miner loop itself leave its operation essentially unchanged.

Reply:
Well... maybe you are right. The original is masked.